client/src/pages/EditProfile.js

```diff
@@ -73,6 +74,10 @@ class EditProfilePage extends Component {
     this.setState({ user });
   };
 
+  handleRateChange = event => {
```

Comment:
Can you add these lines in your handleRateChange?

let user = { ...this.state.user };
user.rate = event.target.value;
this.setState({ user });

then, user's rate state is updated